### PR TITLE
Enable Zendesk tickets from Automatticians

### LINKED ISSUE #21449: Append "+testing" to feedback form email for Automatticians
From Zendesk:

> Attempts to submit requests with a staff member role (admin, agent, etc.) identity will fail by design. Apologies for any confusion around this behavior.

The end result is that HEs are unable to test the feedback form. When they try, they see this:

feedback

According to Zendesk, a potential workaround for this is to programmatically append "-testing" when the feedback form is submitted from an @automattic.com or @a8c.com email address.

**Update:** This change should be made for all Zendesk ticket creation, not just the support form.

[Related iOS issue](https://github.com/wordpress-mobile/WordPress-iOS/issues/23794)

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #21449 

Automatticians who have a Zendesk staff member role (admin, agent, etc.) identity have long been unable to test creating  Zendesk tickets in the apps.

According to Zendesk, a workaround for this is to programmatically append "+testing" when a ticket is submitted from an @automattic.com or @a8c.com email address. This PR does just that.

To test:

* Sign into an account with an Automattic email, or modify [this line](https://github.com/wordpress-mobile/WordPress-Android/blob/312b6da9e1ff89f5b41ea4355829e4d824fa35da/WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java#L897) to use a hard-coded Automattic email
* Verify that the feedback form now works
* Verify that you can now submit support tickets

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java [hunk 1]
```diff
@@ -894,7 +894,15 @@ public static void setSupportEmail(String email) {
     }
 
     public static String getSupportEmail() {
-        return getString(DeletablePrefKey.SUPPORT_EMAIL);
+        String email = getString(DeletablePrefKey.SUPPORT_EMAIL);
+        // Zendesk can't create support tickets for Automattic email addresses from
+        // those with a staff member role (admin, agent, etc.), so insert "-testing"
+        // into the email to make it work
+        if ((email.contains("@automattic.com") || email.contains("@a8c.com")) && !email.contains("-testing@")) {
+            return email.replace("@", "-testing@");
```
Comment (crazytonyli) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java]: I think we are suppose to add an `+testing`, not `-`? /cc @jkmassel

Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java]: @crazytonyli Thanks for catching that! I've made the correction.

Comment (nbradbury) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java]: @crazytonyli Are we okay to merge this or are we waiting for @jkmassel to approve it?

**Review by crazytonyli — APPROVED** [review]

**Pull request merged**